### PR TITLE
Translate "Ruby 3.4.0 RC 1 Released" (ko)

### PR DESCRIPTION
**Pull request by riseshia**
:link: https://github.com/ruby/www.ruby-lang.org/issues/2818

Translates #3435 #3436 #3437

Actual diff is cd2198962

### CODE REVIEW

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+* Ruby의 내장 가비지 컬렉터는 `gc/default/default.c`에 분리되어 있으며,
+  `gc/gc_impl.h`에 정의된 API를 사용하여 Ruby와 상호 작용합니다.
+  내장 가비지 컬렉터는 `make modular-gc MODULAR_GC=default`를 사용하여
+  라이브러리로서 빌드하고 환경 변수 `RUBY_GC_LIBRARY=default`를
+  사용하여 활성화할 수 있습니다. [[Feature #20470]]
+
+* [MMTk](https://www.mmtk.io/)를 기반으로 한 실험적인 GC 라이브러리가 제공됩니다.
+  이 GC 라이브러리는 `make modular-gc MODULAR_GC=mmtk`를 사용하여 빌드하고
+  환경 변수 `RUBY_GC_LIBRARY=mmtk`를 사용하여 활성화할 수 있습니다.
+  이는 빌드 머신에 Rust 도구 체인이 필요합니다. [[Feature #20860]]
```
Comment (yous) [inline ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md:33]: I think "툴체인" has been widely used.

```suggestion
  이는 빌드 머신에 Rust 툴체인이 필요합니다. [[Feature #20860]]
```

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+
+## 언어 변경
+
+* 파일에 `frozen_string_literal` 주석이 없을 때, 문자열 리터럴이 변경되면
+  폐기 예정 경고를 출력합니다.
+  이 경고는 `-W:deprecated`나 `Warning[:deprecated] = true` 설정을 통해 활성화할 수 있습니다.
+  이 변경을 무효화하고 싶다면 Ruby를 실행할 때 `--disable-frozen-string-literal` 커맨드라인 인수를
```
Comment (yous) [inline ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md:41]: There are both "커맨드라인" and "커맨드 라인", so let's use only one of them.
```suggestion
  이 변경을 무효화하고 싶다면 Ruby를 실행할 때 `--disable-frozen-string-literal` 커맨드 라인 인수를
```

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+TL;DR:
+* x86-64와 arm64 플랫폼에서 대부분의 벤치마크에서 성능이 향상되었습니다.
+* YJIT 메타 데이터 컴파일의 메모리 사용량이 줄었습니다.
+* 여러 버그가 수정되었습니다. YJIT는 이제 더 견고하고 잘 테스트되었습니다.
+
+새 기능:
+* YJIT의 메모리 사용량을 추적하는 통합 메모리 제한을 `--yjit-mem-size`
+  커맨드 라인 옵션(기본값 128 MiB)으로 추가했습니다.
+  이는 이전 `--yjit-exec-mem-size`보다 직관적이며, YJIT의 전체 메모리 사용량을 추적합니다.
```
Comment (yous) [inline ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md:64]: How about this?

```suggestion
* `--yjit-mem-size` 커맨드 라인 옵션(기본값 128 MiB)을 통해
  YJIT의 통합 메모리 제한을 추가했습니다.
  이는 이전 `--yjit-exec-mem-size`보다 직관적이며, YJIT의 전체 메모리 사용량을 추적합니다.
```

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+* YJIT 메타 데이터를 저장하는 데 필요한 메모리를 줄이는 콘텍스트 압축.
+* 로컬 변수를 위한 레지스터를 할당할 수 있는 개선된 할당기.
+* YJIT을 사용할 때 Ruby로 작성된 더 많은 코어 프리미티브를 사용합니다.
+  * 성능을 높이기 위해 Ruby로 다시 작성된 `Array#each`, `Array#select`, `Array#map` [[Feature #20182]].
+* 작고 사소한 메서드를 인라인으로 변환하는 능력.
+  * 빈 메서드
+  * 상수를 반환하는 메서드
+  * `self`를 반환하는 메서드
+  * 인수를 직접 반환하는 메서드
+* 더 많은 실행시간 메서드에 대한 특별한 코드 생성
```
Comment (yous) [inline ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md:81]:
```suggestion
* 더 많은 런타임 메서드에 대한 특별한 코드 생성
```

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+* YJIT 메타 데이터 컴파일의 메모리 사용량이 줄었습니다.
+* 여러 버그가 수정되었습니다. YJIT는 이제 더 견고하고 잘 테스트되었습니다.
+
+새 기능:
+* YJIT의 메모리 사용량을 추적하는 통합 메모리 제한을 `--yjit-mem-size`
+  커맨드 라인 옵션(기본값 128 MiB)으로 추가했습니다.
+  이는 이전 `--yjit-exec-mem-size`보다 직관적이며, YJIT의 전체 메모리 사용량을 추적합니다.
+* 이제 `RubyVM::YJIT.runtime_stats`를 통해 항상 더 많은 통계가 사용 가능합니다.
+* `--yjit-log`를 통해 무엇이 컴파일되었는지 추적하는 컴파일 로그를 추가했습니다.
+  * 실행 중에도 로그의 마지막 부분을 `RubyVM::YJIT.log`로 확인할 수 있습니다.
```
Comment (yous) [inline ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md:67]:
```suggestion
  * 런타임에서도 로그의 마지막 부분을 `RubyVM::YJIT.log`로 확인할 수 있습니다.
```

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+* YJIT을 사용할 때 Ruby로 작성된 더 많은 코어 프리미티브를 사용합니다.
+  * 성능을 높이기 위해 Ruby로 다시 작성된 `Array#each`, `Array#select`, `Array#map` [[Feature #20182]].
+* 작고 사소한 메서드를 인라인으로 변환하는 능력.
+  * 빈 메서드
+  * 상수를 반환하는 메서드
+  * `self`를 반환하는 메서드
+  * 인수를 직접 반환하는 메서드
+* 더 많은 실행시간 메서드에 대한 특별한 코드 생성
+* `String#getbyte`, `String#setbyte` 및 다른 문자열 메서드를 최적화
+* 저 레벨 비트/바이트 조작을 빠르게 하기 위한 비트 연산 최적화
```
Comment (yous) [inline ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md:83]:
```suggestion
* 저레벨 비트/바이트 조작을 빠르게 하기 위한 비트 연산 최적화
```

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md (hunk 1 above)
Comment (riseshia) [reply ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md]: 👍

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md (hunk 3 above)
Comment (riseshia) [reply ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md]: Looks me better 👍

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md (hunk 2 above)
Comment (riseshia) [reply ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md]: fixed [1e55b9a](https://github.com/ruby/www.ruby-lang.org/pull/3438/commits/1e55b9aaed21d1e8cc4335b1f85cb75c81f674ca)

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md (hunk 3 above)
Comment (riseshia) [reply ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md]: fixed [1e55b9a](https://github.com/ruby/www.ruby-lang.org/pull/3438/commits/1e55b9aaed21d1e8cc4335b1f85cb75c81f674ca)

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md (hunk 6 above)
Comment (riseshia) [reply ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md]: fixed [1e55b9a](https://github.com/ruby/www.ruby-lang.org/pull/3438/commits/1e55b9aaed21d1e8cc4335b1f85cb75c81f674ca)

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md (hunk 5 above)
Comment (riseshia) [reply ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md]: fixed [1e55b9a](https://github.com/ruby/www.ruby-lang.org/pull/3438/commits/1e55b9aaed21d1e8cc4335b1f85cb75c81f674ca)

File: ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md (hunk 4 above)
Comment (riseshia) [reply ko/news/_posts/2024-12-12-ruby-3-4-0-rc1-released.md]: fixed [1e55b9a](https://github.com/ruby/www.ruby-lang.org/pull/3438/commits/1e55b9aaed21d1e8cc4335b1f85cb75c81f674ca)

**Review by yous — APPROVED** [review]
Thanks!

**Comment by riseshia** [comment]
🙏

**Pull request merged**